src/actions/edit_actions.ts

```diff
@@ -170,31 +170,31 @@ function cannotMerge(env: SpreadsheetChildEnv): boolean {
   const { sheetId } = env.model.getters.getActivePosition();
   const { xSplit, ySplit } = env.model.getters.getPaneDivisions(sheetId);
   return (
-    zones.length > 1 ||
-    (top === bottom && left === right) ||
+    (top === bottom && left === right && zones.length === 1) ||
     (left < xSplit && xSplit <= right) ||
-    (top < ySplit && ySplit <= bottom)
+    (top < ySplit && ySplit <= bottom) ||
+    hasOverlappingZones(zones)
   );
 }
 
-function isInMerge(env: SpreadsheetChildEnv): boolean {
-  if (!cannotMerge(env)) {
-    const zones = env.model.getters.getSelectedZones();
-    const { col, row, sheetId } = env.model.getters.getActivePosition();
-    const zone = env.model.getters.expandZone(sheetId, positionToZone({ col, row }));
-    return isEqual(zones[0], zone);
+function isUnmergeAction(env: SpreadsheetChildEnv): boolean {
+  if (cannotMerge(env)) {
+    return false;
   }
-  return false;
+
+  const firstZone = env.model.getters.getSelectedZones()[0];
+  const sheetId = env.model.getters.getActiveSheetId();
+  const mergesInFirstZone = env.model.getters.getMergesInZone(sheetId, firstZone);
+  return mergesInFirstZone.length > 0;
```

Comment:
you change the current behavior with a single selected zone that contains a merge. Before we would expand the merge to contain all of the the selection, now you'll just remove the merge.

That's a bit strange considering that we do the opposite when there's multiple selections. Not sure if that's what you discussed with Francois

Reply:
Also just checking only if the first zone has merges seems a bit arbitrary... But idk how we can be consistent if we keep the "expand merge" feature...

Maybe if any zone in the selection has a merge we could remove all of the merges, and if there's no merge in the selection we'd merge every zone. That'd be consistent at least, and I think that's what excel does ? And we'd drop the "expand merge" feature" To discuss with Francois.

Reply:
Hi @dhrp-odoo, let's keep what you did. It is seems consistent. Thanks!